### ISSUE TITLE
Update documentation with details about how developers can build products that are inclusive

### ISSUE THREAD

**Issue by bl-aire** [jupyterhub/jupyterhub#4082]
"The Web is fundamentally designed to work for everyone, not minding their hardware, software, language, location, or ability. When a Web product this goal, it is accessible to people with a diverse range of hearing, movement, sight, and cognitive ability"

Unfortunately, not every developer knows or takes the time to implement Web Content Accessibility guidelines (WCAG)

### My idea:

With issue #4078 a means to automate web accessibility testing in the development phase is being developed. I am proposing that a guide on how to set up required tools and run this test is included in the documentation.
The guide can introduce web accessibility briefly with links to sites one can visit to learn more about tools used to test it, how to test the project during development etc 

### Who would gain from this:

Developers working on this project will be able to build having the needs of diverse users in mind and even people who go through this project will be learning about building digital products that are centered on meeting the needs of diverse people.

**Comment by bl-aire** [comment]
Hello @minrk @sgibson91 ,

This I related to #4078 which I am working on. And I would love to work on it.

**Comment by iLynette** [comment]
Hello @bl-aire I would love to work on this one, is it possible?

**Comment by bl-aire** [comment]
> Hello @bl-aire, @minrk, @sgibson91  I would love to work on this one, is it possible?

Hello Lynette,

I am working on it but I can reach out to see how we can look at it together.

**Comment by bl-aire** [comment]
> Hello @bl-aire, @minrk, @sgibson91  I would love to work on this one, is it possible?

Hello Lynette,
I am currently working on this as it is related to an issue I created earlier.

**Comment by iLynette** [comment]
Okay @bl-aire good luck with it😊

**Comment by bl-aire** [comment]
Thank you 🙏

**Comment by sgibson91** [comment]
@bl-aire I think the project would _love_ a guide on which tools to use to test accessibility, what aspects should/can be tested, and how to construct/automate those tests. However, I feel this is a task that can be tackled during the internship itself, rather than the contribution period, if that makes sense?

**Comment by bl-aire** [comment]
> @bl-aire I think the project would _love_ a guide on which tools to use to test accessibility, what aspects should/can be tested, and how to construct/automate those tests. However, I feel this is a task that can be tackled during the internship itself, rather than the contribution period, if that makes sense?

It totally makes sense. I would like to know if creating issues only are accepted as contributions in this project please.

**Comment by manics** [comment]
@bl-aire How familiar are you with the architecture of JupyterHub / JupyterLab / Jupyter-server? The UI component of JupyterHub is relatively small, though is still worth analysing for accessibility (being small means it should be more tractable). If you're interested in the accessibility of the notebook user interfaces that's a separate component- there's some discussions on
https://discourse.jupyter.org/c/special-topics/accessibility/29

**Comment by minrk** [comment]
> I would like to know if creating issues only are accepted as contributions in this project please.

Yes, I think creating clearly expressed issues describing problems to solve and approaches to take absolutely counts as a contribution. Thank you for opening these!

**Comment by bl-aire** [comment]
> > I would like to know if creating issues only are accepted as contributions in this project please.
> 
> 
> 
> Yes, I think creating clearly expressed issues describing problems to solve and approaches to take absolutely counts as a contribution. Thank you for opening these!

Noted

Thank you 🙏

**Comment by trallard** [comment]
hey y'all chiming in here @bl-aire  thanks for your contributions in the area of accessibility (these are much needed and welcome) - I would highly suggest dropping by the jupyter accessibility calls to see what other folks in the community are working on regarding accessibility in the Jupyter space

A few of us (me included) are working on accessibility testing for JupyterLab and Jupyter notebooks already, so would be best to align to avoid duplication of work and instead collaborate and support each other. 

> @bl-aire I think the project would love a guide on which tools to use to test accessibility, what aspects should/can be tested, and how to construct/automate those tests. 

Also, I am unsure if this kind of resource should be "JupyterHub" specific since most of the UI would come from either JupyterLab or Jupyter notebook.
In the spirit of community/project alignment and impact, I think it would make sense to have this into https://github.com/jupyter/accessibility instead. I am happy to help with questions, orientation and anything needed in this space.

**Comment by sgibson91** [comment]
Hi @trallard 👋🏻 

> Also, I am unsure if this kind of resource should be "JupyterHub" specific since most of the UI would come from either JupyterLab or Jupyter notebook.

It definitely shouldn't! Hub has a few pages that aren't from Notebook or Lab (spawn, admin, token) so we should align with the accessibility team on what tools and best practices we should be adopting in Hub repos as a common toolkit once work kicks off in earnest next month.

@bl-aire I endorse Tania's offer to visit the accessibility team if you have the time, they're awesome people doing really important work.

And I commit to do a better job synchronising these efforts moving forward 💜

**Comment by bl-aire** [comment]
> hey y'all chiming in here @bl-aire  thanks for your contributions in the area of accessibility (these are much needed and welcome) - I would highly suggest dropping by the jupyter accessibility calls to see what other folks in the community are working on regarding accessibility in the Jupyter space

I had a look at https://github.com/jupyter/accessibility and the page dedicated to improving project Jupyter's accessibility on the Discourse forum. It helped me understand the project and current measures that have been taken to improve accessibility. I definitely will join the team and contribute in any way I can.

> I am happy to help with questions, orientation and anything needed in this space. 

Thank you so much

**Comment by bl-aire** [comment]
> @bl-aire I endorse Tania's offer to visit the accessibility team if you have the time, they're awesome people doing really important work.

I certainly will. And I am looking forward to learning and contributing in any way I can.
